### ISSUE TITLE
`toml::to_string` serializes `-0.0` to `--0.0` which `toml::from_str` can't parse

### ISSUE THREAD

**Issue by Eijebong** [alexcrichton/toml-rs#462]
```
#[derive(serde::Deserialize, serde::Serialize)]
struct Foo {
    a: f64,
}

fn main() {
    let foo = Foo { a: -0.0 };
    let s = toml::to_string(&foo).unwrap();
    println!("{}", s);
    toml::from_str::<Foo>(&s).unwrap();
}
```

```
a = --0.0

thread 'main' panicked at 'called `Result::unwrap()` on an `Err` value: Error { inner: ErrorInner { kind: Wanted { expected: "newline", found: "a period" }, line: Some(0), col: 7, at: Some(7), message: "", key: [] } }', src/main.rs:10:31
```

**Comment by panicbit** [comment]
It looks like this is fixed on `HEAD`, but is broken in the latest `crates.io` version:

![image](https://user-images.githubusercontent.com/628445/163380644-111b713d-783c-49ac-9a0a-bc3caeefe13f.png)

**Comment by ehuss** [comment]
Thanks for the heads up, I didn't realize this hadn't been published yet.  I just published 0.5.9 which should include the fix for this from #426.
